libs/entitlement/src/main/java/org/elasticsearch/entitlement/initialization/EntitlementInitialization.java

```diff
@@ -21,6 +21,13 @@
 import java.util.Map;
 import java.util.Set;
 
+/**
+ * Called by the agent during {@code agentmain} to configure the entitlement system,
+ * instantiate and configure an {@link EntitlementChecker},
+ * make it available to the bootstrap library via {@link #checker()},
+ * and then install the {@link org.elasticsearch.entitlement.instrumentation.Instrumenter}
+ * to begin injecting our instrumentation.
```

Comment:
The length of this paragraph is a hint that combining the instrumentation and entitlement checking into one library is an awkward fit. The two parts of the system are almost entirely unrelated, have distinct responsibilities, and different dependencies.

The instrumentation:
- is concerned with bytecodes
- depends on ASM
- is the same for every method (modulo argument forwarding)
- has no relation to policies / grants
- is active only during warmup
- has no steady-state throughput impact

The entitlement checking:
- is concerned with policies/grants
- doesn't use ASM
- is different for every method
- has steady-state throughput impact

There's really no overlap here, so combining these overloads one library and gives it multiple responsibilities.

Reply:
I agree with what you say, but I come to different conclusions: we definitely should keep them separate "as if" they were in 2 libraries, but I think keep separate packages/classes is enough. 
It's different if we think we'll have a different implementation (non-ASM), but do you see that happen in the near future?
Anyway, I'm fine with the way it is now too.

Reply:
They ought to be separate modules because they have different dependencies. The entitlement checking module has no need for ASM; the instrumentation module has no need for Jack's policy data structures. They also have a well-defined boundary (`EntitlementChecker`). They can be tested independently.

It's an ideal place to split into two modules with cleanly defined responsibilities. I'm not sure what the hesitation is.